File: src/main/java/io/pages/MainPage.java
package io.pages;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.PageFactory;

import java.util.logging.Level;

import io.appium.java_client.pagefactory.AndroidFindBy;
import io.appium.java_client.pagefactory.AppiumFieldDecorator;
import io.appium.java_client.pagefactory.iOSXCUITFindBy;
import utils.log.Log;

public class MainPage extends CommonPage {

    @AndroidFindBy(id = "com.owncloud.android:id/root_toolbar")
    @iOSXCUITFindBy(xpath = "(//XCUIElementTypeButton[@name=\"Show/Hide sidebar\"])[2]")
    private WebElement mainPage;

    private static final MainPage INSTANCE = new MainPage();

    private MainPage() {
        super();
        PageFactory.initElements(new AppiumFieldDecorator(getDriver()), this);
    }

    //Singletonize
    public static MainPage getInstance() {
        return INSTANCE;
    }

    public boolean isMainPageDisplayed() {
        Log.log(Level.FINE, "Starts: Is Main Page Displayed.");
        return mainPage.isDisplayed();
    }
}

File: src/main/kotlin/io/pages/MainPage.kt
package io.pages

import io.appium.java_client.pagefactory.AndroidFindBy
import io.appium.java_client.pagefactory.AppiumFieldDecorator
import org.openqa.selenium.By
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.PageFactory
import utils.Log
import java.util.logging.Level


class MainPage : CommonPage() {

    @AndroidFindBy(xpath = "//androidx.compose.ui.platform.ComposeView/android.view.View/" +
            "android.view.View/android.view.View/android.view.View[3]/android.widget.Button")
    private lateinit var fabButton: WebElement

    init {
        PageFactory.initElements(AppiumFieldDecorator(getDriver()), this)
    }

    fun isMainPageDisplayed(): Boolean {
        Log.log(Level.FINE, "Checking whether main page is displayed")
        return ::fabButton.isInitialized && fabButton.isDisplayed
    }

    fun isFolderDisplayed(folderName: String): Boolean {
        Log.log(Level.FINE, "Checking whether files are displayed")
        val folder = getDriver()!!.findElement(By.xpath(
            "//android.widget.TextView[@text=\"$folderName\"]"))
        return folder.isDisplayed
    }

    companion object {
        val instance: MainPage by lazy { MainPage() }
    }
}
